### PR TITLE
Shipping method with pickup places - zasilkovna

### PR DESCRIPTION
**Pull request by boris-brtan**
| Q             | A
| ------------- | ---
|Description, reason for the PR| we want to try extensibility of shopsys framework
|New feature| Yes 
|BC breaks| Yes 
|Fixes issues| #3
|Standards and tests pass| Yes/No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by LukasHeinz** [comment]
Please add entry to changelog...

File: tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php [hunk 1]
```diff
@@ -147,16 +147,16 @@ public function testAll(): void
...
-        $this->doTestExtendedEntityInstantiation(OrderPayment::class, ExtendedOrderPayment::class, self::ORDER_PAYMENT_ID);
-        $this->doTestExtendedEntityInstantiation(OrderProduct::class, ExtendedOrderProduct::class, self::ORDER_PRODUCT_ID);
-        $this->doTestExtendedEntityInstantiation(OrderTransport::class, ExtendedOrderTransport::class, self::ORDER_TRANSPORT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderItem::class, ExtendedOrderPayment::class, self::ORDER_PAYMENT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderItem::class, ExtendedOrderProduct::class, self::ORDER_PRODUCT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderItem::class, ExtendedOrderTransport::class, self::ORDER_TRANSPORT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderPayment::class, ExtendedOrderPayment::class, self::ORDER_PAYMENT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderProduct::class, ExtendedOrderProduct::class, self::ORDER_PRODUCT_ID);
+//        $this->doTestExtendedEntityInstantiation(OrderTransport::class, ExtendedOrderTransport::class, self::ORDER_TRANSPORT_ID);
```
Comment (Miroslav-Stopka) [inline tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php]: Why is this code commented out? Is it on purpose? Or just some forgotten change?

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig [hunk 2]
```diff
@@ -16,45 +16,48 @@
...
-                    attr: { class: 'js-checkbox-toggle', 'data-checkbox-toggle-container-id': 'js-company-fields' },
-                    label: 'I buy on company behalf'|trans})
-                }}
-                <div id="js-company-fields">
-                    {{ form_row(form.companyName, { label: 'Company name'|trans }) }}
-                    {{ form_row(form.companyNumber, { label: 'Company number'|trans }) }}
-                    {{ form_row(form.companyTaxNumber, { label: 'Tax number'|trans }) }}
-                </div>
-            </fieldset>
+            <div id="js-company-and-delivery-address-fields">
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Why did you add this new div? Id js-company-and-delivery-address-fields is never used.

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig [hunk 3]
```diff
@@ -16,45 +16,48 @@
...
-                    {{ form_row(form.deliveryCity, { label: 'City'|trans }) }}
-                    {{ form_row(form.deliveryPostcode, { label: 'Postcode'|trans }) }}
-                    {{ form_row(form.deliveryCountry, { label: 'Country'|trans }) }}
-                </div>
-            </fieldset>
+                <fieldset>
+                    {{ form_row(form.deliveryAddressFilled, {
+                        attr: { class: 'js-checkbox-toggle', 'data-checkbox-toggle-container-id': 'js-delivery-address-fields' },
+                        label: 'I want to deliver products to different address than the billing one'|trans,
+                        disabled: pickupPlaceDelivery}) }}
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: pickupPlaceDelivery contains information about the fact if transport is already checked and this checked transport is pickUpPlace type. Am I right? If so, please consider renaming the variable like "isSelectedPickupPlaceTypeTransport". I mean, you should be capable to recognize the content of the variable only by variable name. Don't you think?

File: tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php [hunk 4]
```diff
@@ -2,9 +2,9 @@
 
 namespace Tests\ShopBundle\Database\Model\Transport;
 
-use Shopsys\FrameworkBundle\Model\Transport\Transport;
-use Shopsys\FrameworkBundle\Model\Transport\TransportDataFactory;
-use Shopsys\FrameworkBundle\Model\Transport\TransportFactory;
+use Shopsys\FrameworkBundle\Model\Transport\TransportDataFactoryInterface;
```
Comment (Miroslav-Stopka) [inline tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php:5]: I am kind of confused with your usage of interfaces. In a class `VatFacadeTest` you changed `Shopsys\FrameworkBundle\Model\Transport\TransportDataFactory` => `Shopsys\ShopBundle\Model\Transport\TransportDataFactory`. In this class `TransportDomainTest` you changed `Shopsys\FrameworkBundle\Model\Transport\TransportDataFactory`  => `Shopsys\FrameworkBundle\Model\Transport\ TransportDataFactoryInterface ` . Please, explain it to me? Why  did you choose TransportDataFactoryInterface and sometimes just  TransportDataFactory from ShopBundle?

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js [hunk 5]
```diff
@@ -16,7 +19,9 @@
             'groups': function () {
 
                 var groups = [Shopsys.constant('\\Shopsys\\FrameworkBundle\\Form\\ValidationGroup::VALIDATION_GROUP_DEFAULT')];
-                if ($orderPersonalInfoForm.find('#order_personal_info_form_deliveryAddressFilled').is(':checked')) {
+                var pickUpPlaceTransport = $orderPersonalInfoForm.find('.js-pick-up-place-hide-if-chosen').hasClass('display-none');
+                var deliveryAddressFilled = $orderPersonalInfoForm.find('#order_personal_info_form_deliveryAddressFilled').is(':checked');
+                if (deliveryAddressFilled && !pickUpPlaceTransport) {
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js]: Is this condition part `!pickUpPlaceTransport` necessary? Explain it to me.

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+            Shopsys.pickUpPlaceSelection.showSearchWindow($(this));
+
+            $(this).prop('checked', false);
+            event.stopImmediatePropagation();
+            event.preventDefault();
+        }
+    };
+
+    Shopsys.pickUpPlaceSelection.showSearchWindow = function ($selectedTransportInput) {
+        var pickUpPlaceInput = $('#transport_and_payment_form_pickUpPlace').val();
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js:34]: I think, variable `pickUpPlaceInput` will be always empty or null because id `transport_and_payment_form_pickUpPlace` does not exist. This pickupPlace.js is probably copied.

File: src/Shopsys/ShopBundle/Resources/config/services.yml [hunk 7]
```diff
@@ -29,6 +30,16 @@ services:
 
   Shopsys\FrameworkBundle\Model\Article\ArticleDataFactoryInterface: '@Shopsys\ShopBundle\Model\Article\ArticleDataFactory'
 
+  League\Flysystem\FilesystemInterface: '@main_filesystem'
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/config/services.yml]: I think this is not connected to your US.

File: src/Shopsys/ShopBundle/Resources/config/services.yml [hunk 8]
```diff
@@ -37,4 +48,20 @@ services:
...
+
+  Shopsys\FrameworkBundle\Model\Order\FrontOrderData: '@Shopsys\ShopBundle\Model\Order\FrontOrderData'
+
+  Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
+
+  Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceRepository: ~
+
+  Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceLoader: ~
+
+  Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceFacade: ~
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/config/services.yml]: Do me a favour and order this services of yours alphabetically :)

File: src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * @param \Shopsys\ShopBundle\Model\Transport\TransportData $transportData
+     * @param \Shopsys\ShopBundle\Model\Transport\Transport $transport
+     */
+    protected function fillFromTransport(BaseTransportData $transportData, BaseTransport $transport)
+    {
+        $transportData->type = $transport->getType();
+        parent::fillFromTransport($transportData, $transport);
+    }
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php:52]: What about method `createFromTransport()`?

File: src/Shopsys/ShopBundle/Model/Transport/TransportData.php [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+class TransportData extends BaseTransportData
+{
+    /**
+     * @var string
+     */
+    public $type;
+
+    public function __construct()
+    {
+        parent::__construct();
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Transport/TransportData.php]: I think here should be something like $this->type = someDefaultValue

File: src/Shopsys/ShopBundle/Model/Product/Feed/Google/GoogleFeedItemFactory.php [hunk 11]
```diff
@@ -3,9 +3,6 @@
 namespace Shopsys\ShopBundle\Model\Product\Feed\Google;
 
 use Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig;
-use Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade;
-use Shopsys\FrameworkBundle\Model\Product\Collection\ProductUrlsBatchLoader;
-use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceCalculationForUser;
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Product/Feed/Google/GoogleFeedItemFactory.php]: Just unused imports?

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+     * @param string $searchQuery
+     * @param string[] $countryCodes
+     * @param string $transportType
+     * @return mixed
+     */
+    public function findActiveBySearchQueryAndTransportType($searchQuery, $countryCoodes, $transportType)
+    {
+        return $this->pickUpPlaceRepository->findActiveBySearchQueryAndTransportType(
+            $searchQuery,
+            $countryCoodes,
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php]: Describe me this scenario in which you are trying to find pickUpPlaces by several country codes. I can image try to find by one code.

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceData.php [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public $gpsLongitude;
+    /**
+     * @var string
+     */
+    public $name;
+    /**
+     * @var string
+     */
+    public $description;
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceData.php:46]: Missing some default values settings.

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceCronModule.php [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $this->logger = $logger;
+    }
+
+    public function run()
+    {
+        $this->updatePickupPlaces($this->feedUrl);
+    }
+
+    protected function updatePickupPlaces($feedUrl)
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceCronModule.php]: Annotate

File: src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+        } else {
+            $orderData->companyName = null;
+            $orderData->companyNumber = null;
+            $orderData->companyTaxNumber = null;
+        }
+
+        return $orderData;
+    }
+
+    private function fillPickUpPlaceDeliveryData(OrderData $orderData, FrontOrderData $frontOrderData)
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php:44]: I see, shipping address form is disabled and that is the reason why u need to fill the delivery address from chosen pickupPlace. Am I right? But if you set the shipping address form on readonly instead of disabled, this method will be unnecessary. Correct?

File: src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+        if ($frontOrderData->companyCustomer) {
+            $orderData->companyName = $frontOrderData->companyName;
+            $orderData->companyNumber = $frontOrderData->companyNumber;
+            $orderData->companyTaxNumber = $frontOrderData->companyTaxNumber;
+        } else {
+            $orderData->companyName = null;
+            $orderData->companyNumber = null;
+            $orderData->companyTaxNumber = null;
+        }
+
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php]: What about:
$orderData = new OrderData();
parent:: getOrderDataFromFrontOrderData($orderData);

File: src/Shopsys/ShopBundle/Model/Order/OrderData.php [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+namespace Shopsys\ShopBundle\Model\Order;
+
+use Shopsys\FrameworkBundle\Model\Order\FrontOrderData as BaseFrontOrderData;
+
+class OrderData extends BaseFrontOrderData
+{
+    /**
+     * @var \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlace|null
+     */
+    public $pickUpPlace;
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Order/OrderData.php:12]: Missing constructor with default value.

File: src/Shopsys/ShopBundle/Model/Order/Order.php [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+        ?User $user = null
+    ) {
+        if ($orderData->transport !== null && $orderData->transport->isPickUpPlaceType()) {
+            $this->pickUpPlace = $orderData->pickUpPlace;
+        } else {
+            $this->pickUpPlace = null;
+        }
+        parent::__construct($orderData, $orderNumber, $urlHash, $user);
+    }
+
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Order/Order.php:44]: What about edit() method?

File: CHANGELOG.md [hunk 19]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## Unreleased
 ### Added
 - [#7 - Product now has condition attribute that is displayed on product detail page and is generated in google feed](https://github.com/shopsys/demoshop/pull/7)
+- [#6 - Shipping method with pickup places](https://github.com/shopsys/demoshop/pull/6)
+    - transport, Order entities and forms are extended
+    - PickUpPlace entity is added
+    - cronmodule for download of pick up places is created
```
Comment (Miroslav-Stopka) [inline CHANGELOG.md]: I think the description "Added new shipping method Zasilkovna. Pick up places downloaded by cron." would be enough. Im not sure that "transport, Order entities and forms are extended" is not too detailed for CHANGELOG.md

**Review by Miroslav-Stopka — COMMENTED** [review]
My review is done. It was a little bigger than I expected. Please check it :)

File: migrations-lock.yml [hunk 20]
```diff
@@ -262,3 +262,9 @@
 20180724131407:
     class: Shopsys\ShopBundle\Migrations\Version20180724131407
     skip: false
+20180724105136:
+    class: Shopsys\FrameworkBundle\Migrations\Version20180724105136
+    skip: false
+20180724060204:
+    class: Shopsys\FrameworkBundle\Migrations\Version20180724060204
```
Comment (PetrHeinz) [inline migrations-lock.yml]: Why add two migrations in a single commit?

File: src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php (hunk 16 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php]: parent:: getOrderDataFromFrontOrderData($orderData)
```
public function getOrderDataFromFrontOrderData(FrontOrderData $frontOrderData)
    {
        $orderData = $this->orderDataFactory->create();
```
i tried to use use
```
/** @var \Shopsys\ShopBundle\Model\Order\OrderData $orderData */
$orderData = parent:: getOrderDataFromFrontOrderData($orderData);
```

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php (hunk 12 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php]: The result will be one pickup place from one country but in the process of searching for pickup place it will filter all pickup places where shop is allowed to ship.

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js (hunk 6 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js:34]: it will be generated as part of the TransportAndPaymentFormType in step2.twig.html
```
->create('pickUpPlace', HiddenType::class)
```

File: src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php (hunk 15 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php:44]: after consultation, no change needed

File: tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php (hunk 4 above)
Comment (boris-brtan) [reply tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php:5]: My mistake,
should be the key as in services_test.yml, so Shopsys\FrameworkBundle\Model\Transport\ TransportDataFactoryInterface is correct.

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig (hunk 2 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: removed

File: tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php (hunk 1 above)
Comment (boris-brtan) [reply tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php]: it was a big mistake, forgotten changes after searching for tests failing.

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js (hunk 5 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js]: If the shipping address is filled just from backend system, there is no need for checking whether it is valid.

File: tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php (hunk 1 above)
Comment (Miroslav-Stopka) [reply tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php]: Solved

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig (hunk 2 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Solved

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig (hunk 3 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Solved

File: src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig [hunk 21]
```diff
@@ -30,31 +29,31 @@
...
-                    {{ form_row(form.deliveryLastName, { label: 'Last name'|trans }) }}
-                    {{ form_row(form.deliveryCompanyName, { label: 'Company'|trans }) }}
-                    {{ form_row(form.deliveryTelephone, { label: 'Telephone'|trans }) }}
-                    {{ form_row(form.deliveryStreet, { label: 'Street'|trans }) }}
-                    {{ form_row(form.deliveryCity, { label: 'City'|trans }) }}
-                    {{ form_row(form.deliveryPostcode, { label: 'Postcode'|trans }) }}
-                    {{ form_row(form.deliveryCountry, { label: 'Country'|trans }) }}
-                </div>
-            </fieldset>
+                <fieldset>
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Indentation

File: tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php (hunk 4 above)
Comment (Miroslav-Stopka) [reply tests/ShopBundle/Database/Model/Transport/TransportDomainTest.php:5]: .

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js (hunk 5 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/scripts/frontend/validation/form/order.js]: Im not sure, if checking selection of pickup place by class display-none is the correct way. There must be some other way how you can't decide whether pickup place was chosen or not. An finally change pickUpPlaceTransport to isSelectedPickUpPlaceTransport

File: src/Shopsys/ShopBundle/Resources/config/services.yml (hunk 7 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/config/services.yml]: Solved

File: src/Shopsys/ShopBundle/Resources/config/services.yml (hunk 8 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/config/services.yml]: Solved

File: src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+        $transportData = new TransportData();
+
+        $this->fillNew($transportData);
+
+        return $transportData;
+    }
+
+    /**
+     * @param \Shopsys\ShopBundle\Model\Transport\TransportData $transportData
+     * @param \Shopsys\ShopBundle\Model\Transport\Transport $transport
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php:24]: I think one of this param shoud be return.

File: src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php (hunk 9 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/Transport/TransportDataFactory.php:52]: Almost done :D Because annotation

File: src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js (hunk 6 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Resources/scripts/frontend/order/pickUpPlace.js:34]: Solved

File: src/Shopsys/ShopBundle/Model/Transport/TransportData.php (hunk 10 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/Transport/TransportData.php]: Solved

File: src/Shopsys/ShopBundle/Model/Product/Feed/Google/GoogleFeedItemFactory.php (hunk 11 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/Product/Feed/Google/GoogleFeedItemFactory.php]: Solved

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php (hunk 12 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceFacade.php]: I see, the same functionality as our selectbox for countries.

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceData.php (hunk 13 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceData.php:46]: Solved

File: src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceCronModule.php (hunk 14 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/PickUpPlace/PickUpPlaceCronModule.php]: Solved

File: src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+        if ($frontOrderData->companyCustomer) {
+            $orderData->companyName = $frontOrderData->companyName;
+            $orderData->companyNumber = $frontOrderData->companyNumber;
+            $orderData->companyTaxNumber = $frontOrderData->companyTaxNumber;
+        } else {
+            $orderData->companyName = null;
+            $orderData->companyNumber = null;
+            $orderData->companyTaxNumber = null;
```
Comment (Miroslav-Stopka) [inline src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php]: Almost done, probably this "company" part is not necessary too.

File: src/Shopsys/ShopBundle/Model/Order/OrderData.php (hunk 17 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/Order/OrderData.php:12]: Solved

File: src/Shopsys/ShopBundle/Model/Order/Order.php (hunk 18 above)
Comment (Miroslav-Stopka) [reply src/Shopsys/ShopBundle/Model/Order/Order.php:44]: Solved

File: CHANGELOG.md (hunk 19 above)
Comment (Miroslav-Stopka) [reply CHANGELOG.md]: Solved

**Comment by Miroslav-Stopka** [comment]
Hi @boris-brtan,

tests and review are done. I like it. Thank you for this new feature :)
